<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.7"/>
<title>SubcloneSeeker: SubcloneSeeker Documentation</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
  $(window).load(resizeHeight);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">SubcloneSeeker
   &#160;<span id="projectnumber">1.0.0</span>
   </div>
   <div id="projectbrief">Subclone deconvolution software framework</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.7 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li class="current"><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('index.html','');});
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Macros</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="headertitle">
<div class="title">SubcloneSeeker Documentation </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h1>SubcloneSeeker </h1>
<p>Computational Framework to reconstructing tumor clone structures</p>
<h1>Introduction </h1>
<p>Clonal heterogeneity is a defining feature of many tumor types, and the elucidation of the subclones that exist within the tumor sample, together with the corresponding cellular fractions, is essential to understand tumorigenesis, relapse, and metastasis. We developed a software framework, SubcloneSeeker, that examines somatic variation events (such as copy number changes, loss of heterozygosity, or point mutations) in order to identify the underlying subclone structure, i.e. the subclones including the normal (non-cancerous) cells and their cellular frequencies within the tumor tissue.</p>
<h2>Getting Started </h2>
<p>Please refer to the documentation for how to get started, or run examples.</p>
<p><a href="http://yiq.github.io/SubcloneSeekerDoc/">http://yiq.github.io/SubcloneSeekerDoc/</a></p>
<h2>Directory Structure </h2>
<p>'src' directory contains all the classes that consists of the SubcloneSeeker core library, in which the basic data types and logics for representing somatic mutations, mutation clusters and subclones are defined.</p>
<p>'utils' directory contains the source code for many command-line utilities that utilizing the SubcloneSeeker core library.</p>
<p>'test' directory contains the test cases for the core library.</p>
<p>'doc' directory contains the project's documentation, generated by Doxygen. A pre-built version can be read at <a href="http://yiq.github.io/SubcloneSeekerDoc/">http://yiq.github.io/SubcloneSeekerDoc/</a></p>
<p>'contrib' directory contains third part libraries that the project uses.</p>
<h2>License </h2>
<p>SubcloneSeeker is distributed under MIT license.</p>
<p>The MIT License (MIT)</p>
<p>Copyright (c) 2013 Yi Qiao</p>
<p>Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:</p>
<p>The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.</p>
<p>THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. </p><h1>Command line utilities </h1>
<p>Descriptions and usage about the included command line utilities can be found at the <a href="utils.html">Command line utilities</a> page</p>
<h1>Running the examples </h1>
<p>There are four examples prepared for you in a easy-to-run fashion. Each of the example represents a figure in the article. In order reproduce the figures, make sure you have R installed, and the binary can be found through PATH lookup. If you run the following command </p><pre class="fragment">which R
</pre><p>And see an absolute path, such as '/usr/bin/R', then the requirement is satisfied</p>
<h2>Getting the example tarball </h2>
<p>The example tarball consists of four results presented in the paper. The tarball is at the root directory of the source tree.</p>
<p><a href="https://github.com/yiq/SubcloneSeeker/blob/master/examples.tar.gz">https://github.com/yiq/SubcloneSeeker/blob/master/examples.tar.gz</a></p>
<h2>Setup the variables </h2>
<p>Navigate to the example directory on a commandline. Edit the file 'setup.sh' so that it reflects how you organized the project files. Make sure that you have compiled the binaries in the SubcloneSeeker project. If you haven't done so, please refer to the 'INSTALLATION' section in this document.</p>
<h3>Example-1: Simulation</h3>
<p><b>Directory: 01-simulation</b></p>
<p>The first example simulates 1000 tumors, each with 3, 4, ..., 8 subclones. Subclone Deconvolution is then performed on all the samples, and statistics will be collected as in how many possible solutions for each tumor. You can initiate the simulation by running the 'sim.sh' script </p><pre class="fragment">./sim.sh
</pre><p>Note that this will take some time to finish, especially in the case 7 or 8 subclones per tumor. Results will be stored in 'result' directory. To plot the results, launch R in the example directory and run the 'plot.r' scriot </p><pre class="fragment">R
source('plot.r');
</pre><p>A result set from a run executed on my computer is also provided, if you just want to plot the results. In that case, please use the 'plot_prerun.r' script</p>
<h3>Example-2: Purity estimation with the snuc cancer cell-line dataset</h3>
<p><b>Directory: 02-snuc</b></p>
<p>In this example, a snuc cancer cell line (thus very high purity) was sequenced, and its sequencing reads were combined with those reads from sequencing the paired-normal sample at various ratios. Subclone deconvolution is performed on each of the digitally created samples, and purity estimation is reported. In order to run the example, simple type </p><pre class="fragment">make
</pre><p>in the example directory, as it will take care of removing old results (if you run it before), creating required directories, running the deconvolution and plotting the result. After the execution, a file 'SamplePurity.pdf' should be created in the same directory, which contains the plotted result.</p>
<h3>Example-3: Re-analysis of the dataset presented in Ding et, al. 2012</h3>
<p>execute </p><pre class="fragment">make
</pre><p>As an example, the following is part of the output </p><pre class="fragment">SAMPLE UPN933124:
Primary tree 11 is compatible with Secondary tree 1
</pre><p>This means that, for the sample UPN933124, the primary tree, with a root node whose id is 11, is compatible with the relapse (secondary) tree, with a root node whose id is 1. To further investigate the actual structures, the utility <code>treeprint</code> can be used. The following commands assume that an environment variable <code>SSHOME</code> exists that points to the directory of SubcloneSeeker </p><pre class="fragment">$SSHOME/utils/treeprint -l results/subclones/UPN933124-pri.sqlite
1
6
11
16
21
26
</pre><p>There are 6 subclone structures (or root nodes, more precisely) in the database resulted from reconstruction on the primary sample. Since tree 11 is compatible, let's look at that one particularly </p><pre class="fragment">$SSHOME/utils/treeprint -r 11 results/subclones/UPN933124-pri.sqlite
0,(0.127401,(0.531157,0.29044,(0.051003,)))
</pre><p>It indicates that the structure of tree 11 is as follows:</p>
<ul>
<li>The root node, which represent normal tissue, is 0% (because the data was adjusted by the highest AF before subjected to reconstruction)</li>
<li>The next subclone, whose parent is the normal clone, has a subclone frequency of 12.74% (let's denote it as subclone A)</li>
<li>The next two subclones are both the children of subclone A, each taking up 53.11% and 29.04% of the entire population (denoted as subclone B and C)</li>
<li>The next subclone, whose parent is subclone C, takes up 5.1% of the entire population</li>
</ul>
<p>A more visualization friendly output can be produced with the <code>-g</code> option with <code>treeprint</code>, which print the structure in graphviz format </p><pre class="fragment">$SSHOME/utils/treeprint -r 11 -g results/subclones/UPN933124-pri.sqlite
digraph {
    n11 [label="n11: 0%"];
    n12 [label="n12: 12.7%"];
    n13 [label="n13: 53.1%"];
    n14 [label="n14: 29%"];
    n15 [label="n15: 5.1%"];
    n11-&gt;n12;
    n12-&gt;n13;
    n12-&gt;n14;
    n14-&gt;n15;
}
</pre><p>Currently the clusters are not labeled when they are provided to the main reconstruction algorithm, so that the output cannot be labeled either (a future update will remedy this), and they are simply labeled as n (as in node) + their subclone ID. But given the subclone frequencies, it is easy to assign the mutations back to each of the subclones, working from the bottom up.</p>
<h3>Example-4: Analysis of two OvCa cancer sample Single Nucleotide Polymorphism dataset</h3>
<p>execute </p><pre class="fragment">make
</pre><p>The details of the resulting structures can be interrogated in a similar fashion as in the previous example </p>
</div></div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated on Thu Jul 3 2014 19:31:15 for SubcloneSeeker by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.7 </li>
  </ul>
</div>
</body>
</html>
